src/main/java/com/puppycrawl/tools/checkstyle/checks/design/InnerTypeLastCheck.java

```diff
@@ -95,8 +114,7 @@ public void visitToken(DetailAST ast) {
             DetailAST nextSibling = ast.getNextSibling();
             while (nextSibling != null) {
                 if (!ScopeUtil.isInCodeBlock(ast)
-                    && (nextSibling.getType() == TokenTypes.VARIABLE_DEF
-                        || nextSibling.getType() == TokenTypes.METHOD_DEF)) {
+                    && CLASS_MEMBER_TOKENS.contains(nextSibling.getType())) {
```

Comment:
@strkkk this can be `!TokenUtil.isTypeDeclaration(nextSibling.getType())` after #8345

Also, `if (!ScopeUtil.isInCodeBlock(ast)` should be moved out of the loop.
Moreover, the loop itself is redundant. Only the next sibling should be checked. If the next sibling is a type, it will be processed soon.

Reply:
1. Do you think this PR should wait until #8345?
2. If loop is removed, it will cause changes in amount of violation generated by check.
I think violation should be raised on type declaration ast, but if it is raised on class members token, it should be all of them and loop is correct here.

Reply:
I'm ok to merge this PR as is.